samples/client/petstore/python/petstore_api/model_utils.py

```diff
@@ -186,6 +187,24 @@ def __getattr__(self, attr):
         """get the value of an attribute using dot notation: `instance.attr`"""
         return self.__getitem__(attr)
 
+    def __copy__(self):
+        cls = self.__class__
+        return cls._new_from_openapi_data(**self.__dict__)
```

Comment:
Why is copy assuming a from server context? Why not use self._spec_property_naming to pick if we should use `__new__` or `_new_from_openapi_data`?
This same question applies to __deepcopy__

Reply:
If I have an object, I would assume that should be able to copy it as is, without consideration for read-only args/etc. Now I can't think of a case where that would occur, but I'm not sure I have a grasp of all the ways objects can be created at moment. That was my reasoning, but if you still feel the logic you have described should be used, I will update to that. thanks.

Reply:
You will be able to copy an instance of the model that is made using client side data, and you can do the sam with an instance of the model made with data from the server, but those contexts should allow slightly different data to be stored in the instances. Changing _spec_property_naming changes where the model says its data came from, which we should not be doing.
Please make the suggested change, thanks.

Reply:
I have updated based on your recommendation.

A question regarding the the petstore_api tests, how exactly are those run? I typically run those manually, but I don't see a CI job that actually runs those?

Reply:
They were run by travis, I need to switch them over to CircleCi because we hit a free limit usage on Travis.
Merging this is blocked by that fix.
Thank you for making the changes.